tests/replicate_test.go

```diff
@@ -203,3 +329,35 @@ func TestReplication(t *testing.T) {
 		}
 	}
 }
+
+func TestReplicationMultipeer(t *testing.T) {
+	if os.Getenv("WITH_GOLEAK") == "1" {
+		defer goleak.VerifyNone(t,
+			goleak.IgnoreTopFunction("github.com/syndtr/goleveldb/leveldb.(*DB).mpoolDrain"),           // inherited from one of the imports (init)
+			goleak.IgnoreTopFunction("github.com/ipfs/go-log/writer.(*MirrorWriter).logRoutine"),       // inherited from one of the imports (init)
+			goleak.IgnoreTopFunction("github.com/libp2p/go-libp2p-connmgr.(*BasicConnMgr).background"), // inherited from github.com/ipfs/go-ipfs/core.NewNode
+			goleak.IgnoreTopFunction("github.com/jbenet/goprocess/periodic.callOnTicker.func1"),        // inherited from github.com/ipfs/go-ipfs/core.NewNode
+			goleak.IgnoreTopFunction("github.com/libp2p/go-libp2p-connmgr.(*decayer).process"),         // inherited from github.com/ipfs/go-ipfs/core.NewNode)
+			goleak.IgnoreTopFunction("go.opencensus.io/stats/view.(*worker).start"),                    // inherited from github.com/ipfs/go-ipfs/core.NewNode)
+		)
+	}
+
+	for _, amount := range []int{
+		2,
+		5,
+		6,
+		//8,  //FIXME: need improve "github.com/libp2p/go-libp2p-pubsub to completely resolve problem
```

Comment:
For 8-nodes, 10-nodes tests
The test is passed if no other programs running on my PC.
To run these tests completely passed, need to increase buffer size in github.com/libp2p/go-libp2p-pubsub.
Because processing in WatchMessages are impacted when too much of go-routines are executed in 8-nodes/10-nodes tests.